nsqd/http.go

```diff
@@ -114,6 +114,51 @@ func (s *httpServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	s.router.ServeHTTP(w, req)
 }
 
+func (s *httpServer) checkAuth(req *http.Request, cmd, topicName, channelName string) error {
+	var isAuthEnabled func() bool
+	var tls int32
+	if req.TLS != nil {
+		isAuthEnabled = s.ctx.nsqd.IsAuthEnabledHTTPS
+		tls = 1
+	} else {
+		isAuthEnabled = s.ctx.nsqd.IsAuthEnabledHTTP
+		tls = 0
+	}
+
+	if isAuthEnabled() {
```

Comment:
style nitpick, but let's invert this and return early so the primary code path is less indented.